sw/device/lib/testing/test_framework/freertos_hooks.c

```diff
@@ -2,18 +2,31 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-#include "sw/device/lib/testing/test_framework/freertos_hooks.h"
-
+#include "sw/device/lib/irq.h"
 #include "sw/device/lib/runtime/hart.h"
 #include "sw/device/lib/runtime/log.h"
 #include "sw/vendor/freertos_freertos_kernel/include/FreeRTOS.h"
+#include "sw/vendor/freertos_freertos_kernel/include/task.h"
 
 // NOTE: the function names below do NOT, and cannot, conform to the style
 // guide, since they are specific implementations of FreeRTOS defined functions.
 
+/**
+ * This is called if configUSE_MALLOC_FAILED_HOOK is set to 1 in
+ * FreeRTOSConfig.h, and a call to pvPortMalloc() fails.
+ */
 void vApplicationMallocFailedHook(void) {
-  // TODO: communicate this event back to the host.
-  LOG_INFO("Malloc Failed.");
-  taskDISABLE_INTERRUPTS();
+  LOG_INFO("FreeRTOS malloc failed.\nIncrease heap size in FreeRTOSConfig.h");
```

Comment:
LOG doesn't handle newlines great; I think you can just do away with them.

Reply:
Fixed.